pkg/apiserver/jsonpatch/jsonpatch.go

```diff
@@ -34,6 +34,10 @@ func (p *PatchSet) WithTest(path string, value interface{}) *PatchSet {
 	return p
 }
 
+func (p *PatchSet) IsEmpty() bool {
```

Comment:
outside the diff, but having addOperation reorder the patches makes it impossible to chain multiple mutation options together with preconditions that depend on prior mutations.  If we remove the reordering, then we keep the easy path easy and make the hard path possible.  If we keep the reordering, the hard path is impossible.

Reply:
I just checked the following command on my local cluster and it works:
```
k patch nodestatuses.example.com sample-nodestatus --type='json' -p='[
  {
    "op": "test",
    "path": "/spec/nodeStatuses/0/nodeName",
    "value": "node1"
  },
  {
    "op": "remove",
    "path": "/spec/nodeStatuses/0"
  },
  {
    "op": "test",
    "path": "/spec/nodeStatuses/0/nodeName",
    "value": "node2"
  },
  {
    "op": "remove",
    "path": "/spec/nodeStatuses/0"
  }
]'
```

Reply:
so the `test` operations don't have to be at the beginning as we originally thought.